app/src/views/OperationalLearning/Sources/index.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+    Button,
+    Container,
+    type ContainerProps,
+    List,
+    Pager,
+} from '@ifrc-go/ui';
+import {
+    useBooleanState,
+    useTranslation,
+} from '@ifrc-go/ui/hooks';
+import { numericIdSelector } from '@ifrc-go/ui/utils';
+import { isDefined } from '@togglecorp/fujs';
+
+import Link from '#components/Link';
+import useFilterState from '#hooks/useFilterState';
+import {
+    type GoApiResponse,
+    useRequest,
+} from '#utils/restRequest';
+
+import AllExtractsModal from './AllExtractsModal';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+type AppealDocumentResponse = GoApiResponse<'/api/v2/appeal_document/'>;
+type AppealDocument = NonNullable<AppealDocumentResponse['results']>[number];
+
+type Props = {
+    summaryType: 'sector' | 'component' | 'insight';
+    summaryId: number;
+}
+
+const PAGE_SIZE = 10;
+
+function Sources(props: Props) {
+    const {
+        summaryId,
+        summaryType,
+    } = props;
+
+    const strings = useTranslation(i18n);
+
+    const [showExtractsModal, {
+        setTrue: setShowExtractsModalTrue,
+        setFalse: setShowExtractsModalFalse,
+    }] = useBooleanState(false);
+
+    const {
+        page: appealDocumentActivePage,
+        setPage: setAppealDocumentActivePage,
+        limit: appealDocumentLimit,
+        offset: appealDocumentOffset,
+    } = useFilterState<object>({
+        filter: {},
+        pageSize: PAGE_SIZE,
+    });
+
+    const {
+        pending: appealDocumentPending,
+        response: appealDocumentResponse,
+        error: appealDocumentError,
+    } = useRequest({
+        url: '/api/v2/appeal_document/',
+        query: {
+            insight_component_id: summaryType === 'component' ? summaryId : undefined,
+            insight_sector_id: summaryType === 'sector' ? summaryId : undefined,
+            insight_id: summaryType === 'insight' ? summaryId : undefined,
+            limit: appealDocumentLimit,
+            offset: appealDocumentOffset,
+        },
+        preserveResponse: true,
+    });
+
+    const appealRendererParams = (_: number, appealDocument: AppealDocument): ContainerProps => ({
+        childrenContainerClassName: styles.appeal,
+        children: (
+            <>
+                <Link
+                    to="emergencyDetails"
+                    urlParams={{
+                        emergencyId: appealDocument.appeal.event.id,
+                    }}
+                    withUnderline
+                >
+                    {appealDocument.appeal?.event.name}
+                </Link>
+                {isDefined(appealDocument?.document_url) && (
+                    <Link
+                        href={appealDocument?.document_url}
+                        withLinkIcon
+                        external
+                    >
+                        {appealDocument?.name}
+                    </Link>
+                )}
+            </>
+        ),
+    });
```

Comment:
Passing components will make the component re-render.
We should create a separate component instead.
cc: @frozenhelium @samshara